duneggd/SubDetector/SandECal.py

```diff
@@ -8,13 +8,36 @@
 
 class SandECalBuilder(gegede.builder.Builder):
     #^~^~^~^~^~^~^~^~^~^~^~^~^~^~^~^~^~^~^~^~^~^~^~^~^~^~^~^~^~^~^~^~^~^~^~^
-    def configure(self, **kwds):
-        self.NCaloModBarrel = 24
+    def configure(self, NCaloModBarrel=None, **kwds):
+        self.NCaloModBarrel = NCaloModBarrel
         self.caloThickness = Q('23cm')
         self.EndcapZ = Q('1.69m')
         self.EndcapRmin = Q('20.8cm')
         self.BarrelRmin = Q('2m')
         self.BarrelDZ = Q('2.15m')
+        self.ECCurvRad = Q('10cm')
+        self.ECStraight = Q('20cm')
+        self.AlPlateThick= Q('2.5cm')
+
+        ### Below values should be the same as the one in SANDBuilder
+        ### and will be used to construct subtraction boolean volume
+        ### to avoid overlap with yoke endcap elements (C,D)
+        
+        # part C is a TUBS, 2.15<|x|<2.58m, rmin=0.84m, rmax=1.34m
+        self.EndcapCZStart=Q("2.15m")
+        self.EndcapCZEnd=Q("2.58m")
+        self.EndcapCRmax=Q("1.34m")
+        self.EndcapCRmin=Q("0.84m")
+        
+        # part D is a TUBS, 1.96<|x|<2.15m, rmin=0.512m, rmax=1.73m
+        self.EndcapDZStart=Q("1.96m")
+        self.EndcapDZEnd=Q("2.15m")
+        # Here we reduce the size of the D element of the yoke endcap
+        # to avoid overlap with the curved element of the ecal endcap module
+        # self.EndcapDRmax=Q("1.73m")
+        # self.EndcapDRmin=Q("0.51m")
```

Comment:
Pleas do not include any commented out lines without explanation what they are for or why they are commented out. When in doubt, just remove them.